### ISSUE TITLE
generate-git-snapshot fails if repo is detached.

### ISSUE THREAD

**Issue by heftig** [mika/jenkins-debian-glue#161]
When building with the branch specifier set to a commit ID, Jenkins detaches the HEAD and sets `GIT_BRANCH` to `detached`.

The `git checkout -f ${GIT_BRANCH}` at the end of generate-git-snapshot fails because a branch named "detached" does not exist.

**Comment by mika** [comment]
Uhm, well, that's not really supported, though I understand that this has its use-case.

Any suggestions how to properly solve that for you?

**Comment by heftig** [comment]
Perhaps `git checkout -f ${GIT_BRANCH} || git checkout -f ${GIT_COMMIT}` ?

**Comment by mika** [comment]
Could you please test whether this would fix your issue?

**Comment by heftig** [comment]
Testing this now, will report back:

```sh
# switch back to previous "branch" before removing the tmp branch
if ! git checkout -f "${GIT_BRANCH}"; then
  # we were probably detached
  git checkout -f "${GIT_COMMIT}"
fi
```

**Comment by heftig** [comment]
Looks like it works. :-)

**Comment by mika** [comment]
I've just merged your fix, thanks a lot for your bug report, feedback + patch! :+1:

**Comment by mika** [comment]
FTR, fix was with commit 737e4e950ae4b0e5a44, I took the wrong github issue number by mistake :-/
